arci-ros2/tests/test_navigation.rs

```diff
@@ -0,0 +1,197 @@
+#![cfg(feature = "ros2")]
+
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use arci::*;
+use arci_ros2::{r2r, Ros2Navigation};
+use assert_approx_eq::assert_approx_eq;
+use futures::{
+    future::{self, Either},
+    stream::{Stream, StreamExt},
+};
+use r2r::nav2_msgs::action::NavigateToPose;
+
+#[tokio::test]
+async fn test_nav() {
+    const ACTION_NAME: &str = "/test_nav";
+
+    let ctx = r2r::Context::create().unwrap();
+    let nav = Ros2Navigation::new(ctx.clone(), ACTION_NAME);
+    let node = Arc::new(Mutex::new(
+        r2r::Node::create(ctx, "nav2_node", "arci_ros2").unwrap(),
```

Comment:
better to change the node name to avoid kill other tests.

```suggestion
        r2r::Node::create(ctx, "test_nav2_node1", "arci_ros2").unwrap(),
```

FYI: namespace "arci_ros2" does not work correctly in r2r now.

Reply:
done